### ISSUE TITLE
How to get the items of a list

### ISSUE THREAD

**Issue by herzog0** [pynamodb/PynamoDB#912]
Hi! 
I'v searched both on the docs and on the examples (and on the issues) but I still have no clue on how to iterate over the items that are stored under a ListAttribute attribute of my class. For example: 
 
 ```python
 class MyClass(Model):
    class Meta:
        table_name = dynamo_table

    chat_id = UnicodeAttribute(hash_key=True)
    entries = ListAttribute(default=[])
    
    def get_entries(self) -> list:
        <what should go here?>
 ```

**Comment by CloudySnake** [comment]
Some more detail on what you're trying to do would be good, but I'll a take a guess that you've written some data, and you need to query it and iterate through the entries. 

So I would re-read the section on querying as a start point
https://pynamodb.readthedocs.io/en/latest/quickstart.html#querying

This will show you how to return data from Dynamo.  Once you have a row returned you can access it with the dot notation so "my_row.entries" will give the list of entries.

As a side note, unless Pynamo is doing something magical under the hood (it might be, I'm afraid I don't know) I would not set the default value as a list.  Setting the default value of a function to a mutable can lead to unexpected behaviour (like I said though, it's possible Pynamo allows for this, in which case hopefully someone more knowledgeable than I will come along to correct me).

**Comment by ikonst** [comment]
It doesn't do anything magical, so the usual [mutable defaults caveat](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) applies. Instead you can simply do `ListAttribute(default=list)`.

**Comment by CloudySnake** [comment]
Would simply allowing null=True be better? Would Dynamo end up storing an empty list (with the small but real associated cost of write / storage) or would it not get written to then DB?

**Comment by ikonst** [comment]
IMO not worth the optimization when you take into account you're billed 1 RCU at the minimum. Perhaps it makes some difference in storage cost. If you're going to extremes, you can also give attributes abbreviated "DB names":
```python
first_name = UnicodeAttribute(attr_name='fn')
```

**Comment by CloudySnake** [comment]
Fair point - I've resisted the whole "DB Name" thing so my team don't think I've gone totally crazy, and with a typical DynamoDB bill of $100 per month we're hardly under pressure to optimise costs.
